packages/components/button/src/button.tsx

```diff
@@ -48,28 +54,35 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       form={form}
       onClick={onClick}
       ref={ref}
+      type={type}
     >
       <InnerButton
         alignItems="center"
         borderRadius="4px"
         colorScheme={colorScheme}
-        disabled={disabled}
+        disabled={disabled || submitting}
         height={HEIGHT_MAPPING[size]}
         px={size}
         {...props}
       >
-        {iconPosition === "left" && (
-          <Box display="flex" ml="-xxs" mr="xs">
-            {icon}
-          </Box>
-        )}
-        <Heading as="span" color="inherit" size="s" textDecoration="none">
-          {children}
-        </Heading>
-        {iconPosition === "right" && (
-          <Box display="flex" ml="xs" mr="-xxs">
-            {icon}
-          </Box>
+        {type == "submit" && submitting ? (
+          <LoadingIndicator size="12px" strokeSize={1} color="text.tertiary" />
+        ) : (
```

Comment:
Do you run into any issues with button height/width changing size when submitting? If so you could absolutely position the loading indicator at toggle the visibility style so the elements are always taking up space.

Reply:
There's a slight flicker when it a form doesn't take long to submit, but otherwise it felt ok.